core/src/Kokkos_SYCL_Space.hpp

```diff
@@ -173,6 +260,56 @@ class SharedAllocationRecord<Kokkos::Experimental::SYCLDeviceUSMSpace, void>
                             bool detail = false);
 };
 
+template <>
+class SharedAllocationRecord<Kokkos::Experimental::SYCLSharedUSMSpace, void>
+    : public SharedAllocationRecord<void, void> {
+ private:
+  using RecordBase = SharedAllocationRecord<void, void>;
+
+  SharedAllocationRecord(const SharedAllocationRecord&) = delete;
```

Comment:
We should delete the move operations too (Rule of 5).

Reply:
Done.